### PR TITLE
Completed implementation of Cocktail.

### PR DESCRIPTION
**Pull request by NStro03**

### CODE REVIEW

File: src/components/Cocktail.js [hunk 1]
```diff
@@ -1,11 +1,20 @@
 import React from 'react'
 import { Link } from 'react-router-dom'
 
-const Cocktail = () => {
+const Cocktail = ({ idDrink, strDrink, strAlcoholic, strDrinkThumb, strGlass }) => {
```
Comment (Ishaan-flyfin) [inline src/components/Cocktail.js:4]: why these are named with `str` as prefix?

File: src/components/SearchForm.js [hunk 2]
```diff
@@ -1,11 +1,34 @@
...
+      <form className='search-form' onSubmit={handleSubmit}>
+        <div className='form-control'>
+          <label htmlFor='search'>search your favorite cocktail</label>
+          <input
+            type='text'
+            name='search'
+            id='search'
+            ref={searchRef}
+            value={searchText}
+            onChange={() => setSearchText(searchRef.current.value)}
```
Comment (Ishaan-flyfin) [inline src/components/SearchForm.js:27]: avoid using anonymous functions

File: src/pages/SingleCocktail.js [hunk 3]
```diff
@@ -1,13 +1,79 @@
...
+  const { cocktailList, isLoading, setIsLoading } = useGlobalContext();
+  const [currentCocktail, setCurrentCocktail] = useState({});
+
+  useEffect(() => {
+    let cocktail = cocktailList.find((cocktail) => cocktail["idDrink"] === id);
+
+    async function fetchAndSetCocktail() {
+      let response = await fetch(url + id);
+      const cocktails = await response.json();
+      setCurrentCocktail(cocktails['drinks'] ? cocktails['drinks'][0] : undefined);
```
Comment (Ishaan-flyfin) [inline src/pages/SingleCocktail.js]: we can use optional [chaining here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

File: src/pages/SingleCocktail.js [hunk 4]
```diff
@@ -1,13 +1,79 @@
...
+          </p>
+          <p>
+            <span className='drink-data'>Glass: </span>{currentCocktail["strGlass"]}
+          </p>
+          <p>
+            <span className='drink-data'>Instructions: </span>{currentCocktail["strInstructions"]}
+          </p>
+          <p>
+            <span className='drink-data'>Ingredients: </span>
+            {ingList.map((ing, index) => {
```
Comment (Ishaan-flyfin) [inline src/pages/SingleCocktail.js]: by reading `ingList` name does not tell me what exactly it contains
the naming could be better

File: src/pages/SingleCocktail.js [hunk 5]
```diff
@@ -1,13 +1,79 @@
...
+  }, [])
+
+  if (isLoading) {
+    return <Loading />
+  }
+  if (!currentCocktail) {
+    return (<h2>No cocktails found</h2>);
+  }
+
+  const ingList = [
```
Comment (Ishaan-flyfin) [inline src/pages/SingleCocktail.js]: this should be handled within reducer/context
that is why we are using it to handle complex data
While accessing i should be getting it in a list from state

**Review by Ishaan-flyfin — CHANGES_REQUESTED** [review]
minor comments

File: src/components/Cocktail.js (hunk 1 above)
Comment (NStro03) [reply src/components/Cocktail.js:4]: These were the keys in the Response JSON, so used the same. @Ishaan-flyfin

File: src/components/SearchForm.js (hunk 2 above)
Comment (NStro03) [reply src/components/SearchForm.js:27]: Here I had to send an argument, so had to use an anonymous function to do that. @Ishaan-flyfin

File: src/pages/SingleCocktail.js (hunk 3 above)
Comment (NStro03) [reply src/pages/SingleCocktail.js]: Oh yes, I will change it.

File: src/pages/SingleCocktail.js (hunk 4 above)
Comment (NStro03) [reply src/pages/SingleCocktail.js]: Will change it to IngredientList

File: src/components/Cocktail.js (hunk 1 above)
Comment (Ishaan-flyfin) [reply src/components/Cocktail.js:4]: okay but here also the context comes into play to make keys readable

File: src/components/SearchForm.js (hunk 2 above)
Comment (Ishaan-flyfin) [reply src/components/SearchForm.js:27]: in this case need to create a new named function

**Pull request merged**